lib/plugins/esbuild/index.js

```diff
@@ -273,15 +275,47 @@ class Esbuild {
     return undefined
   }
 
-  _buildProperties() {
+  async _buildProperties() {
```

Comment:
Can we add some sort of memoization/caching of this function? Right now we'd be importing that module multiple times per run when we should only be doing the import once.

Reply:
Hmmm wouldn't that have some side effects in dev mode though? If I make changes to that file while in an active dev mode session, these changes won't be picked up due to the memoization.

Reply:
We could have different behavior for dev mode. I'm just concerned that we are re-importing potentially large files for no reason in these cases.

Reply:
I understand. Ok I'll memoize it.